chainerx_cc/chainerx/python/routines.cc

```diff
@@ -369,6 +369,21 @@ void InitChainerxLinalg(pybind11::module& m) {
             [](const ArrayBodyPtr& a, float rcond) { return MoveArrayBody(PseudoInverse(Array{a}, rcond)); },
             "a"_a,
             "rcond"_a = 1e-15);
+    mlinalg.def(
+            "eigh",
+            [](const ArrayBodyPtr& a, const std::string& UPLO) {
+                std::tuple<Array, Array> wv = Eigh(Array{a}, UPLO.c_str()[0]);
```

Comment:
We must validate `UPLO` here. Let's in the binding check if it's a string of length 1, and in the routine check that it's actually `'U'` or `'L'`.